test/Microsoft.ML.Tokenizers.Tests/DoubleArrayTrieTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Microsoft.ML.Tokenizers.Tests
+{
+    public class DoubleArrayTrieTests
+    {
+        private static readonly (string Key, int Value)[] _entries = new[]
+        {
```

Comment:
Can you find any other interesting test cases from 
https://github.com/google/sentencepiece/blob/d8f741853847553169444afc12c00f4bbff3e9ce/python/test/sentencepiece_test.py
Or
https://github.com/google/sentencepiece/blob/d8f741853847553169444afc12c00f4bbff3e9ce/src/unigram_model_test.cc

These cases here seem pretty simple - was wondering if there were other interesting ones to add.